src/Controllers/RegistrationController.php

```diff
@@ -99,7 +99,7 @@ private function renderSignUpPage(): Response
                 'isPasswordEnabled' => $this->userFactory->determineIsPasswordEnabled(),
                 'isDECTEnabled' => $this->config->get('enable_dect'),
                 'isShowMobileEnabled' => $this->config->get('enable_mobile_show'),
-                'isGoodieEnabled' => $goodieType !== GoodieType::None,
+                'isGoodieEnabled' => ($goodieType !== GoodieType::None) && config('enable_email_goody'),
```

Comment:
```suggestion
                'isGoodieEnabled' => $goodieType !== GoodieType::None && config('enable_email_goody'),
```
does not spark joy ;)